src/main/java/com/minecolonies/coremod/commands/ColonyTPCommand.java

```diff
@@ -0,0 +1,176 @@
+package com.minecolonies.coremod.commands;
+
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyManager;
+import com.minecolonies.coremod.configuration.Configurations;
+import com.minecolonies.coremod.util.ServerUtils;
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentString;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.common.FMLCommonHandler;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+
+import static com.minecolonies.coremod.MineColonies.findLand;
+import static com.minecolonies.coremod.MineColonies.isPositionSafe;
+import static com.minecolonies.coremod.commands.AbstractSingleCommand.Commands.COLONYTP;
+
+/**
+ * this command is made to TP a player to a safe random spot that is not to close to another colony
+ * Need to add a configs permissions check
+ * Need to allow OPs to send players ./mc ctp (Player) if player is not allowed
+ */
+public class ColonyTPCommand extends AbstractSingleCommand
+{
+    public static final  String DESC = "ctp";
+    private static final int ATTEMPTS = Configurations.numberOfAttemptsForSafeTP;
+    private static final int UPPER_BOUNDS = Configurations.maxDistanceFromWorldSpawn * 2;
+    private static final int LOWER_BOUNDS = Configurations.maxDistanceFromWorldSpawn;
+    private static final int SPAWN_NO_TP = Configurations.minDistanceFromWorldSpawn;
+    private static final int STARTING_Y = 250;
+    private static final double ADDS_TWENTY_PERCENT = 1.20;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "ADDS_TWENTY_PERCENT" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1068)